src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+
+    // CONSIDER: we really don't need these
+    int256 private maxDisplayPrice;
+    int256 private minDisplayPrice;
+
+    // CONSIDER: figure out approprate values for these
+    uint256 private constant AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT = 11 * 10**20;
+    uint256 private constant LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**21;
+    uint256 private constant ALL_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**22;
+
+    uint256 private constant MIN_NUM_OUTCOMES = 2;
+    uint256 private constant MIN_PAYOUT_DENOMINATOR = 2;
+    uint256 private constant MAX_FEE_PER_ETH_IN_ATTOETH = 5 * 10 ** 17;
+    int256 private constant DISPLAY_PRICE_MIN = -(2 ** 254);
+    int256 private constant DISPLAY_PRICE_MAX = 2 ** 254;
+    uint256 private constant SET_COST_MULTIPLIER_MAX = 2 ** 254;
+    uint256 private constant APPROVAL_AMOUNT = 2 ** 254;
+    uint256 private constant AUTOMATED_REPORTING_DURATION_SECONDS = 3 days;
+    uint256 private constant AUTOMATED_REPORTING_DISPUTE_DURATION_SECONDS = 3 days;
+    address private constant NULL_ADDRESS = address(0);
+
+    ReportingWindow private reportingWindow;
+    uint256 private endTime;
+    uint8 private numOutcomes;
+    uint256 private marketCreationBlock;
+    bytes32 private topic;
+    address private automatedReporterAddress;
+    mapping(bytes32 => ReportingToken) private reportingTokens;
+    Cash private denominationToken;
+    address private creator;
+    IShareToken[] private shareTokens;
+    uint256 private finalizationTime;
+    bool private automatedReportReceived;
+    bytes32 private tentativeWinningPayoutDistributionHash;
+    bytes32 private finalPayoutDistributionHash;
+    DisputeBondToken private automatedReporterDisputeBondToken;
+    DisputeBondToken private limitedReportersDisputeBondToken;
+    DisputeBondToken private allReportersDisputeBondToken;
+    uint256 private validityBondAttoeth;
+    uint256 private automatedReporterBondAttoeth;
+
+    function initialize(ReportingWindow _reportingWindow, uint256 _endTime, uint8 _numOutcomes, uint256 _payoutDenominator, uint256 _feePerEthInAttoeth, Cash _denominationToken, address _creator, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, bytes32 _topic) public payable beforeInitialized returns (bool _success) {
+        endInitialization();
+        require(address(_reportingWindow) != NULL_ADDRESS);
+        require(_payoutDenominator >= MIN_PAYOUT_DENOMINATOR);
+        require(_numOutcomes >= MIN_NUM_OUTCOMES);
+        // payoutDenominator must be a multiple of numOutcomes so we can evenly split complete set share payout on indeterminate
+        require((_payoutDenominator % _numOutcomes) == 0);
+        require(feePerEthInAttoeth <= MAX_FEE_PER_ETH_IN_ATTOETH);
+        require((DISPLAY_PRICE_MIN <= _maxDisplayPrice) && (_maxDisplayPrice <= DISPLAY_PRICE_MAX));
+        require((DISPLAY_PRICE_MIN <= _minDisplayPrice) && (_minDisplayPrice <= DISPLAY_PRICE_MAX));
```

Comment:
I don't believe these two are necessary anymore now that we have real types.  display price can be any int256 but we had the check previously I think out of habit because Serpent didn't have types so everything was confusing around the edges (this particular check probably wasn't necessary in Serpent either).